core/controllers/profile.py

```diff
@@ -317,41 +317,74 @@ def put(self) -> None:
         update_type = self.payload.get('update_type')
         data = self.payload.get('data')
         bulk_email_signup_message_should_be_shown = False
-        if update_type == 'user_bio':
-            if len(data) > feconf.MAX_BIO_LENGTH_IN_CHARS:
-                raise self.InvalidInputException(
-                    'User bio exceeds maximum character limit: %s'
-                    % feconf.MAX_BIO_LENGTH_IN_CHARS)
+        user_settings = user_services.get_user_settings(self.user_id)
 
-            user_services.update_user_bio(self.user_id, data)
-        elif update_type == 'subject_interests':
-            user_services.update_subject_interests(self.user_id, data)
+        # Here we use type Any because we don't know the Input type.
+        def is_data_of_non_valid_type(
```

Comment:
Name `is*` gives impression that it returns a boolean. I'd call it "validate_data" or "validate_data_type"

There's also no point to define it inside this function. The more nested the code the more difficult it is to understand it. Just create a private function of the class.

Reply:
Done.